PINCE.py

```diff
@@ -1895,36 +1872,34 @@ def removeOffsetLayout(self):
         frame.deleteLater()
         self.verticalLayout_Pointers.removeWidget(frame)
         del self.offsetsList[-1]
-        del self.offsetDerefLabels[-1]
-        del self.offsetTextLabels[-1]
         self.update_value()
 
     def update_deref_labels(self, pointer_chain_result: typedefs.PointerChainResult):
         if pointer_chain_result != None:
             base_deref = hex(pointer_chain_result.pointer_chain[0]).upper().replace("X","x")
-            self.label_BaseAddressDeref.setText(f"-> {base_deref}")
-            for index, derefLabel in enumerate(self.offsetDerefLabels):
-                previousDerefValue = pointer_chain_result.pointer_chain[index]
-                if previousDerefValue == 0:
-                    previousDerefText = "<font color=red>??</font>"
+            self.label_BaseAddressDeref.setText(f" → {base_deref}")
+            for index, offsetFrame in enumerate(self.offsetsList):
+                previousDerefText = self.caps_hex_or_error_indicator(pointer_chain_result.pointer_chain[index])
+                currentDerefText = self.caps_hex_or_error_indicator(pointer_chain_result.pointer_chain[index + 1])
+                offsetText = self.caps_hex(offsetFrame.offsetText.text())
+                operationalSign = "" if offsetText.startswith("-") else "+"
+                calculation = f"{previousDerefText}{operationalSign}{offsetText}"
+                if index != len(self.offsetsList) - 1:
+                    offsetFrame.update_deref_label(f" [{calculation}] → {currentDerefText}")
                 else:
-                    previousDerefText = hex(previousDerefValue).upper().replace("X","x")
+                    offsetFrame.update_deref_label(f" {calculation} = {currentDerefText}")
+        else:
+            self.label_BaseAddressDeref.setText(" → <font color=red>??</font>")
+            for offsetFrame in self.offsetsList:
+                offsetFrame.update_deref_label(" → <font color=red>??</font>")
 
-                currentDerefValue = pointer_chain_result.pointer_chain[index+1]
-                if currentDerefValue == 0:
-                    currentDerefText = "<font color=red>??</font>"
-                else:
-                    currentDerefText = hex(currentDerefValue).upper().replace("X","x")
+    def caps_hex_or_error_indicator(self, hex_int:int):
```

Comment:
slight nitpick but offset list contains non-hex ints, so we might wanna change the naming of this param

Reply:
renamed to address, since its exactly that.